### PR TITLE
Mobile Friendly Flatpickr Cal On Admin.

### PR DESCRIPTION
**Pull request by MatthewKennedy**
Strip out jQuery UI calendar and replace it with a mobile friendly flatpickr.js

The original date picker throws the iOS keyboard up for no reason when you click to pick a date.

- Backwards compatible with any extensions using the .datepicker css class on the input text field to trigger a calendar.

### CODE REVIEW

File: core/spree_core.gemspec [hunk 1]
```diff
@@ -32,6 +32,7 @@ Gem::Specification.new do |s|
   s.add_dependency 'carmen', '>= 1.0', '< 1.2'
   s.add_dependency 'cancancan', '~> 3.0'
   s.add_dependency 'ffaker', '~> 2.9'
+  s.add_dependency 'flatpickr', '~> 4.6.0'
```
Comment (damianlegawiec) [inline core/spree_core.gemspec:35]: @MatthewKennedy please move this into `spree_backend.gemspec`

**Review by damianlegawiec — COMMENTED** [review]
Awesome work as always @MatthewKennedy :) Could you take a look at CodeClimate and one comment that I've left? Thank you!

**Comment by MatthewKennedy** [comment]
Sure I’ll sort that issue with the gem in the wrong file, and get looking at some of the code climate issue tonight.

**Review by kaliszKrzysztof — APPROVED** [review]

**Comment by MatthewKennedy** [comment]
Still working on this, got some issues with the test.

**Pull request merged**